client/src/components/LocationMap.tsx

```diff
@@ -138,6 +147,160 @@ export function LocationPickerMap(props: {
   );
 }
 
+export function EventsMapView(props: { posts: PostUserPreview[] }) {
+  const [googleMap, setMap] = React.useState(
+    {} as {
+      map: google.maps.Map;
+      maps: typeof google.maps;
+      markers: google.maps.Marker[];
+    }
+  );
+
+  // kinda hacky, but we need to explicitly define height/width for google maps in sx
+  const theme = useTheme();
+  const smQuery = useMediaQuery(theme.breakpoints.between('sm', 'md'));
+  const bgQuery = useMediaQuery(theme.breakpoints.between('md', 'lg'));
+  const xbgQuery = useMediaQuery(theme.breakpoints.up('lg'));
+  const mapHeight = smQuery
+    ? '65vh'
+    : bgQuery
+    ? '90vh'
+    : xbgQuery
+    ? '90vh'
+    : '30vh';
+
+    const mapWidth = smQuery
+    ? '95vh'
+    : bgQuery
+    ? '100vh'
+    : xbgQuery
+    ? '159vh'
+    : '40vh';
+
+  // We want to do manual refresh so that the user is not interrupted when interacting the map if the data is changed/new posts are fetched
+  const refreshMap = () => {
+    if (googleMap.map) {
+      setupMarkers(googleMap.map, googleMap.maps);
+    }
+  };
+
+  // enable additional options (i.e streetview)
+  const getOptions = (maps: GoogleMapReact.Maps) => {
+    return {
+      mapTypeControl: true,
+      streetViewControl: true,
+      fullScreenControl: true,
+      fullSCreenControlOptions: true,
+      styles: [
+        {
+          featureType: 'poi',
+          elementType: 'labels',
+          stylers: [{ visibility: 'on' }],
+        },
+      ],
+    };
+  };
+
+  const clearMarkers = () => {
+    if (!googleMap.map) {
+      return;
+    }
+
+    for (let i = 0; i < googleMap.markers.length; i++) {
+      googleMap.markers[i].setMap(null);
+    }
+
+    setMap({ ...googleMap, markers: [] as google.maps.Marker[] });
+  };
+
+  const setupMarkers = (map: google.maps.Map, maps: typeof google.maps) => {
+    clearMarkers();
+    let markers = [] as google.maps.Marker[];
+
+    for (let i = 0; i < props.posts.length; i++) {
+      const curr = props.posts[i];
+      const { lat, lng } = curr.coords;
+      if (lat !== -1 && lng !== -1) {
+        // show markers for in-person events
+
+        const hasAttendance =
+          curr.capacity > 0
+            ? `<p> Capacity: ${curr.usersCheckedIn}/${curr.capacity}</p>`
+            : `<div></div>`;
+
+        const tmpInfo = new google.maps.InfoWindow({
+          content: `<div><h2 style='word-break: break-all'>${curr.title.slice(0, 100)}...</h2> 
+          <p style='word-break: break-all'> ${curr.body.slice(0, 120) + '...'} </p>
+          <p> Located @ ${curr.location} </p> 
+          ${hasAttendance}
+          <a href="/${curr.id}"> Read More </a>
+          </div>`,
+        });
+
+        const percentFilled =
+          (curr.capacity > 0 ? (curr.usersCheckedIn / curr.capacity) : 1) * 100;
+          
+        // red marker indicates event is full, blue is almost filled, green is almost empty event (less than 50%)
```

Comment:
Shouldn't the marker colours relate to the type of the event? If I'm looking for events I'm not looking for just any event which has capacity, I'd be looking for something specific (type).

Just a thought.

Reply:
The color coding based on capacity was suggested by Ilir.  We could change the icon itself (while maintaining different color codes) for the type of event?

Reply:
Ya, different icons make sense as well + colours.

Reply:
this is fine, only posts of type 'event' has locations for now